src/ape/api/accounts.py

```diff
@@ -315,6 +337,8 @@ def check_signature(
             data = encode_defunct(text=data)
         elif isinstance(data, int):
             data = encode_defunct(hexstr=HexBytes(data).hex())
+        elif isinstance(data, bytes) and (len(data) != 32 or recover_using_eip191):
```

Comment:
small perf, maybe, for short-circuit eval on the `or` stmt

```suggestion
        elif isinstance(data, bytes) and (recover_using_eip191 or len(data) != 32):
```

as checking a bool is prolly faster than a length check